src/datasources/workspace/WorkspaceDataSource.ts

```diff
@@ -14,26 +15,36 @@ export class WorkspaceDataSource extends DataSourceBase<WorkspaceQuery> {
 
   baseUrl = this.instanceSettings.url + '/niuser/v1';
 
-  defaultQuery = {
-    constant: 3.14,
-  };
-
-  async runQuery(query: WorkspaceQuery, { range }: DataQueryRequest): Promise<DataFrameDTO> {
-    return {
-      refId: query.refId,
-      fields: [
-        { name: 'Time', values: [range.from.valueOf(), range.to.valueOf()], type: FieldType.time },
-        { name: 'Value', values: [query.constant, query.constant], type: FieldType.number },
-      ],
-    };
+  defaultQuery = {};
+
+  async runQuery(_query: WorkspaceQuery): Promise<DataFrameDTO> {
+    return { fields: this.workspacesToFields( await this.getWorkspaces()) };
   }
 
-  shouldRunQuery(query: WorkspaceQuery): boolean {
+  shouldRunQuery(): boolean {
     return true;
   }
 
   async testDatasource(): Promise<TestingStatus> {
     await this.get(this.baseUrl + '/workspaces');
     return { status: 'success', message: 'Data source connected and authentication successful!' };
   }
+
+  async metricFindQuery(): Promise<MetricFindValue[]> {
+    const workspaces = await this.getWorkspaces();
+    return workspaces.map(ws => ({ text: ws.name, value: ws.id }));
+  }
+
+  private workspacesToFields(workspaces: Workspace[]): Array<{ name: string, values: string[] }> {
+    const ids: string[] = [];
+    const names: string[] = [];
+    workspaces.forEach((workspace: Workspace) => {
+      ids.push(workspace.id);
+      names.push(workspace.name);
+    });
+    return [
+      { name: 'ID', values: ids },
+      { name: 'Name', values: names }
```

Comment:
Probably not super important, but I think we've been consistent about using lowercase for field names:
https://github.com/ni/systemlink-grafana-plugins/blob/ef51b3aae45e502419d798ff3d910d4d9d9fc8f1/src/datasources/system/SystemDataSource.ts#L41-L53